test/integration/camera_sensor.cc

```diff
@@ -2268,3 +2268,121 @@ TEST_F(CameraSensor, CheckNewAndLegacyDistortionModes)
+    return;
+  }
+
+  // get the 'default_intrinsics_camera' sensor, no <intrinsics> tag is used for this sensor
+  // and we expect the intrinsics to be equal to the default intrinsic values provided by gazebo.
+  std::string defaultIntrinsicsCameraName = "default_intrinsics_camera_sensor";
+  sensors::SensorPtr defaultIntrinsicsSensor = sensors::get_sensor(defaultIntrinsicsCameraName);
+  sensors::CameraSensorPtr defaultIntrinsicsCamSensor =
+      std::dynamic_pointer_cast<sensors::CameraSensor>(defaultIntrinsicsSensor);
+  EXPECT_TRUE(defaultIntrinsicsCamSensor != nullptr);
+  rendering::CameraPtr defaultIntrinsicsCam = defaultIntrinsicsCamSensor->Camera();
+  EXPECT_TRUE(defaultIntrinsicsCam != nullptr);
+
+  // image size, focal length and optical centre for 'default_intrinsics_camera' sensor
+  unsigned int defaultCamWidth = defaultIntrinsicsCam->ImageWidth();
+  unsigned int defaultCamHeight = defaultIntrinsicsCam->ImageHeight();
+  {
+    double defaultCamFx = defaultIntrinsicsCam->ImageFocalLengthX();
+    double defaultCamFy = defaultIntrinsicsCam->ImageFocalLengthY();
+    double defaultCamCx = defaultIntrinsicsCam->ImageOpticalCentreX();
+    double defaultCamCy = defaultIntrinsicsCam->ImageOpticalCentreY();
+
+    EXPECT_GT(defaultCamWidth, 0u);
+    EXPECT_GT(defaultCamHeight, 0u);
+    EXPECT_EQ(defaultCamWidth, 320u);
+    EXPECT_EQ(defaultCamHeight, 240u);
+    double error = 0.0005;
+    EXPECT_NEAR(defaultCamFx, 277.127, error);
+    EXPECT_NEAR(defaultCamFy, 277.127, error);
+    EXPECT_DOUBLE_EQ(defaultCamCx, 160);
+    EXPECT_DOUBLE_EQ(defaultCamCy, 120);
+  }
+
+  // get the 'intrinsics_camera' sensor, <intrinsics> tag is used explicitly for this sensor
+  // where the intrinsics provided are same as gazebo default.
+  std::string intrinsicsCameraName = "intrinsics_camera_sensor";
+  sensors::SensorPtr intrinsicsSensor = sensors::get_sensor(intrinsicsCameraName);
+  sensors::CameraSensorPtr intrinsicsCamSensor =
+      std::dynamic_pointer_cast<sensors::CameraSensor>(intrinsicsSensor);
+  EXPECT_TRUE(intrinsicsCamSensor != nullptr);
+  rendering::CameraPtr intrinsicsCam = intrinsicsCamSensor->Camera();
+  EXPECT_TRUE(intrinsicsCam != nullptr);
+
+  // image size, focal length and optical centre for 'intrinsics_camera_sensor' sensor
+  unsigned int camWidth = intrinsicsCam->ImageWidth();
+  unsigned int camHeight = intrinsicsCam->ImageHeight();
+  {
+    double camFx = intrinsicsCam->ImageFocalLengthX();
+    double camFy = intrinsicsCam->ImageFocalLengthY();
+    double camCx = intrinsicsCam->ImageOpticalCentreX();
+    double camCy = intrinsicsCam->ImageOpticalCentreY();
+
+    EXPECT_GT(camWidth, 0u);
+    EXPECT_GT(camHeight, 0u);
+    EXPECT_EQ(camWidth, 320u);
+    EXPECT_EQ(camHeight, 240u);
+    EXPECT_DOUBLE_EQ(camFx, 277);
+    EXPECT_DOUBLE_EQ(camFy, 277);
+    EXPECT_DOUBLE_EQ(camCx, 160);
+    EXPECT_DOUBLE_EQ(camCy, 120);
+  }
+
+  // connect to new frame event
```

Comment:
I'm not sure what we're testing below this line. IIUC, we're checking that the both sensors return the same image. The opposite might actually be more interesting: demonstrating that changes to the intrinsics does in fact change the image.

Reply:
I will add a unit test for the different images as well.